File: MasterErp.Angular/src/app/components/Purchases/models/PurchaseQuotationModel.ts
import { CreatorModel } from "../../Shared/models/CreatorModel";

export interface PurchaseQuotationModel extends CreatorModel {
    purchaseQuotationId: number;
    quotationNumber: number;
    quotationDate: string;
    notes: string;
    isLocked: boolean | null;
    isCancelled: boolean | null;
    quotationDetails: PurchaseQuotationDetailsModel[];
    totalCount: number | null;
}

export interface PurchaseQuotationDetailsModel {
    purchaseQuotationDetailsId: number;
    purchaseQuotationId: number;
    itemId: number;
    price: number;
    supplierId: number | null;
    supplierNameAR: string;
    supplierNameEN: string;
    notes: string;
    itemNameAR: string;
    itemNameEN: string;
    unitId: number | null;
    unitNameAR: string;
    unitNameEN: string;
}

File: RetailStation.Entities/DTOs/Purchases/PurchaseQuotationModel.cs
using MasterErp.Entities.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MasterErp.Entities.DTOs.Purchases
{
    public class PurchaseQuotationModel : CreatorModel
    {
        public int? PurchaseQuotationId { get; set; }
        public int? QuotationNumber { get; set; }
        public DateTime? QuotationDate { get; set; }
        public string Notes { get; set; }
        public string SerialNumber { get; set; }
        public bool? IsLocked { get; set; }
        public bool? IsCancelled { get; set; }
        public List<PurchaseQuotationDetailsModel> QuotationProducts { get; set; }
        public int? TotalCount { get; set; }


        public int? PurchaseQuotationDetailsId { get; set; }
        public int ItemId { get; set; }
        public double Price { get; set; }
        public int? SupplierId { get; set; }
        public string SupplierNameAR { get; set; }
        public string SupplierNameEN { get; set; }
        public string ItemNameAR { get; set; }
        public string ItemNameEN { get; set; }
        public int? UnitId { get; set; }
        public string UnitNameAR { get; set; }
        public string UnitNameEN { get; set; }
    }
}
